lib/config/defaults.js

```diff
@@ -921,7 +922,11 @@ const applyOutputDefaults = (
 	F(output.environment, "forOf", () => tp && optimistic(tp.forOf));
 	F(output.environment, "bigIntLiteral", () => tp && tp.bigIntLiteral);
 	F(output.environment, "dynamicImport", () => tp && tp.dynamicImport);
-	F(output.environment, "module", () => tp && tp.module);
+	F(
+		output.environment,
+		"module",
+		() => tp && conditionallyOptimistic(tp.module, output.module)
```

Comment:
```suggestion
		() => output.module || (tp && tp.module)
```

Reply:
not sure that this is correct, target could be explicitly not support es modules. e.g. `target: "node 10"`. In my opinion, In this case webpack should throw an error.